### PR TITLE
[PRM-1023] - Added API1811 model

### PR DESCRIPTION
**Pull request by LewisMitchell**

### CODE REVIEW

**Review by ruthdas03 — COMMENTED** [review]
So are we not validating the properties in the model and would want it to be validated at the parser. Also do we not want to include failureResponse model .

**Comment by LewisMitchell** [comment]
It depends what properties you are talking about? I'm not sure whether it is worth having the failure response, we could log out the response body at the parser but we aren't doing anything specific with the failure (i.e. if its a non-200 we would most likely show a technical difficulties page to the user anyway)?

**Comment by ruthdas03** [comment]
> It depends what properties you are talking about? I'm not sure whether it is worth having the failure response, we could log out the response body at the parser but we aren't doing anything specific with the failure (i.e. if its a non-200 we would most likely show a technical difficulties page to the user anyway)?

For example , the property 'periodKey' in 'financialDetailsType' should have maxLength of 4 which for now is not being validated. ie we can have periodKey of length more than 4 and it's valid.
             periodKey:
            description: Period Key
            type: string
            minLength: 1
            maxLength: 4

**Comment by LewisMitchell** [comment]
I don't think validating each property is a great idea given the amount of properties there are. I don't see the need to apply logic which should be checked before it reaches us.

**Review by ruthdas03 — COMMENTED** [review]
Looks ok, just one observation  in case class DocumentDetailsMetadata

File: app/models/v2/financialDetails/DocumentDetails.scala [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                                    paymentLotItem: Option[String],
+                                    interestRate: Option[BigDecimal],
+                                    interestFromDate: Option[LocalDate],
+                                    interestEndDate: Option[LocalDate],
+                                    latePaymentInterestID: Option[String],
+                                    latePaymentInterestAmount: Option[BigDecimal],
+                                    lpiWithDunningBlock: Option[BigDecimal],
+                                    accruingPenaltyLPP1: Option[String]
+                                    //NOTE: commented out due to 22-limit on case class parameters with JSON serialiser
+//                                    lpp1Amount: Option[BigDecimal],
```
[the remark this answers is not in the document]
Comment (LewisMitchell) [reply app/models/v2/financialDetails/DocumentDetails.scala:79]: The success response does have those fields? We may use them in the future but they are not useful to us. If they were to become useful then we would move them out of the metadata model and into the main model. The metadata model is there so we can access that extra data _if_ we need to

**Review by ruthdas03 — APPROVED** [review]

**Pull request merged**